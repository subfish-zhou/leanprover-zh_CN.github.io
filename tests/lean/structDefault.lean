structure A where
  f : Nat â†’ Nat â†’ Nat

structure B extends A where
  f (a b : Nat) := 10

structure C extends A where
  f a b := 0

structure D extends A where
  f a b := true -- error
