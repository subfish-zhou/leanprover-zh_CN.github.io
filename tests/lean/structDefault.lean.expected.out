structDefault.lean:11:11-11:15: error: type mismatch
  true
has type
  Bool : Type
but is expected to have type
  Nat : Type
